exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -945,97 +1684,291 @@ func (e *Exchange) GetHistoricCandles(ctx context.Context, pair currency.Pair, a
+		if err != nil {
+			return nil, err
+		}
+		resp[i] = futures.Contract{
+			Name:                 cp,
+			Type:                 ct,
+			Exchange:             e.Name,
+			SettlementCurrencies: currency.Currencies{settleCurr},
+			MarginCurrency:       settleCurr,
+			Asset:                assetType,
+			StartDate:            contracts[i].ListingDate.Time(),
+			IsActive:             contracts[i].Status == "OPEN",
+			Status:               contracts[i].Status,
+			MaxLeverage:          contracts[i].Leverage.Float64(),
+			SettlementType:       futures.Linear,
+		}
+	}
+	return resp, nil
 }
 
 // GetLatestFundingRates returns the latest funding rates data
-func (e *Exchange) GetLatestFundingRates(context.Context, *fundingrate.LatestRateRequest) ([]fundingrate.LatestRateResponse, error) {
-	// TODO: implement with API upgrade
-	return nil, common.ErrNotYetImplemented
+func (e *Exchange) GetLatestFundingRates(ctx context.Context, r *fundingrate.LatestRateRequest) ([]fundingrate.LatestRateResponse, error) {
+	if r == nil || *r == (fundingrate.LatestRateRequest{}) {
+		return nil, fmt.Errorf("%w LatestRateRequest", common.ErrEmptyParams)
+	}
+	var pairString string
+	if !r.Pair.IsEmpty() {
+		is, err := e.IsPerpetualFutureCurrency(r.Asset, r.Pair)
+		if err != nil {
+			return nil, err
+		} else if !is {
+			return nil, fmt.Errorf("%w %s %v", futures.ErrNotPerpetualFuture, r.Asset, r.Pair)
+		}
+		pairString = r.Pair.String()
+	}
+	contracts, err := e.GetFuturesHistoricalFundingRates(ctx, pairString, time.Time{}, time.Time{}, 0)
+	if err != nil {
+		return nil, err
+	}
+	timeChecked := time.Now()
+	resp := make([]fundingrate.LatestRateResponse, 0, len(contracts))
+	for i := range contracts {
+		cp, err := currency.NewPairFromString(contracts[i].Symbol)
+		if err != nil {
+			return nil, err
+		}
+		var isPerp bool
+		isPerp, err = e.IsPerpetualFutureCurrency(r.Asset, cp)
+		if err != nil {
+			return nil, err
+		} else if !isPerp {
+			continue
+		}
+		rate := fundingrate.LatestRateResponse{
+			Pair:     cp,
+			Exchange: e.Name,
+			Asset:    r.Asset,
+			LatestRate: fundingrate.Rate{
+				Time: contracts[i].FundingRateSettleTime.Time(),
+				Rate: decimal.NewFromFloat(contracts[i].FundingRate.Float64()),
+			},
+			TimeOfNextRate: contracts[i].NextFundingTime.Time(),
+			TimeChecked:    timeChecked,
+		}
+		if r.IncludePredictedRate {
+			rate.PredictedUpcomingRate = fundingrate.Rate{
+				Time: contracts[i].NextFundingTime.Time(),
+				Rate: decimal.NewFromFloat(contracts[i].NextPredictedFundingRate.Float64()),
+			}
+		}
+		resp = append(resp, rate)
+	}
+	return resp, nil
+}
+
+// IsPerpetualFutureCurrency ensures a given asset and currency is a perpetual future
+func (e *Exchange) IsPerpetualFutureCurrency(a asset.Item, cp currency.Pair) (bool, error) {
+	switch {
+	case a == asset.Futures && strings.HasSuffix(cp.Quote.String(), "PERP"):
+		return true, nil
+	default:
+		return false, nil
+	}
```

Comment:
```suggestion
	return a == asset.Futures && strings.HasSuffix(cp.Quote.String(), "PERP"):
```

Could also do cp.Quote.Equal(perpCode) instead of suffix check if its only a static value.